de.metas.business/src/main/java/de/metas/payment/api/impl/AbstractPaymentDAO.java

```diff
@@ -66,6 +68,18 @@ public I_C_Payment getById(@NonNull final PaymentId paymentId)
 		return load(paymentId, I_C_Payment.class);
 	}
 
+	@Override
+	public Optional<I_C_Payment> getByExternalOrderId(@NonNull final ExternalId externalId)
+	{
+		final I_C_Payment i_c_payment = Services.get(IQueryBL.class)
```

Comment:
similar to what i wrote elsewhere about getting C_BPartners by externalId: please also put the OrgId into the query